client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -57,11 +58,24 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep }: Sidebar
 	const site = useSite();
 	const clipboardButtonEl = useRef< HTMLButtonElement >( null );
 	const [ clipboardCopied, setClipboardCopied ] = useState( false );
+	const { isFetchingGlobalStylesStatus, globalStylesInUse, shouldLimitGlobalStyles } =
+		usePremiumGlobalStyles( site?.ID );
 
 	const { flowName, title, launchTitle, subtitle } = getLaunchpadTranslations( flow );
 	const arrayOfFilteredTasks: Task[] | null = getArrayOfFilteredTasks( tasks, flow );
 	const enhancedTasks =
-		site && getEnhancedTasks( arrayOfFilteredTasks, siteSlug, site, submit, goToStep, flow );
+		site && ! isFetchingGlobalStylesStatus
+			? getEnhancedTasks(
+					arrayOfFilteredTasks,
+					siteSlug,
+					site,
+					submit,
+					goToStep,
+					flow,
+					globalStylesInUse,
+					shouldLimitGlobalStyles
```

Comment:
Is there a reason we are using the GS hook here instead of directly into the `getEnhancedTasks`?

Something like this:
```js
case 'unlock_styles':
  const { isFetchingGlobalStylesStatus, globalStylesInUse, shouldLimitGlobalStyles } =
    usePremiumGlobalStyles( site?.ID );

  if (
    isFetchingGlobalStylesStatus ||
    ! shouldLimitGlobalStyles ||
    ! globalStylesInUse
  ) {
    return;
  }

  // render GS task
  break;
```


I feel a bit uncomfortable that we are preventing the whole task list to show up until the GS status is fetched.

Reply:
`getEnhancedTasks` is not a React component, so we cannot use the hook there.

> I feel a bit uncomfortable that we are preventing the whole task list to show up until the GS status is fetched.

I know, but otherwise this is what happens:


https://user-images.githubusercontent.com/1233880/198274092-c980d23d-baee-41e9-9b6a-66a640db5eef.mov